proxy/txnmanager/mod.go

```diff
@@ -0,0 +1,55 @@
+package txnmanager
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/dedis/d-voting/contracts/evoting"
+)
+
+// Manager defines the public HTTP API of the transaction manager
+type Manager interface {
+	// GET /transactions/{token}
+	IsTxnIncluded(http.ResponseWriter, *http.Request)
+
+	// submit the transaction to the blockchain
+	// return the transactionID and 
+	// the index of the last block when it was submitted
+	SubmitTxn(ctx context.Context, cmd evoting.Command, cmdArg string, payload []byte) ([]byte, uint64, error)
+
+	// create the json to send to the 
+	CreateTransactionInfoToSend(txnID []byte, lastBlockIdx uint64, status TransactionStatus) (TransactionInfoToSend, error)
+	SendTransactionInfo(w http.ResponseWriter, txnID []byte, lastBlockIdx uint64, status TransactionStatus) error
+}
+
+// TransactionStatus is the status of a transaction
+type TransactionStatus byte
+
+const (
+	// UnknownTransactionStatus is the basic status of a transaction
+	UnknownTransactionStatus TransactionStatus = 0
+	// IncludedTransaction is the status of a transaction that has been included
+	IncludedTransaction TransactionStatus = 1
+	// RejectedTransaction is the status of a transaction will never be included
+	RejectedTransaction TransactionStatus = 2
+)
+
+// TransactionInfo defines the information of a transaction
```

Comment:
`TransactionInfo` is an internal structure only needed by the transaction manager. As such, it should be private.

Reply:
Also, I would rename it to `transactionInternalInfo`.